src/packages/block/block-type/components/input-block-type/input-block-type.element.ts

```diff
@@ -88,19 +125,62 @@ export class UmbInputBlockTypeElement<BlockType extends UmbBlockTypeBase = UmbBl
 	}
 
 	deleteItem(contentElementTypeKey: string) {
-		this._items = this._items.filter((x) => x.contentElementTypeKey !== contentElementTypeKey);
+		this.value = this._items.filter((x) => x.contentElementTypeKey !== contentElementTypeKey);
 		this.dispatchEvent(new UmbChangeEvent());
 	}
 
 	protected getFormElement() {
 		return undefined;
 	}
 
+	#renameGroup(e: UUIInputEvent, key: string) {
+		if (!key) return;
+		const groupName = e.target.value as string;
+		const groups = this.groups.map((group) => (group.key === key ? { ...group, name: groupName } : group));
+
+		this.#context?.setPropertyValue('blockGroups', groups);
+	}
+
+	#deleteGroup(key: string) {
+		this.#context?.setPropertyValue(
+			'blockGroups',
+			this.groups.filter((group) => group.key !== key),
+		);
+		this.value = this.value.filter((block) => block.groupKey !== key);
+		this.dispatchEvent(new UmbChangeEvent());
+		this.#mapValues();
+	}
+
 	render() {
-		return html`
-			${this._items ? repeat(this._items, (item) => item.contentElementTypeKey, this.#renderItem) : ''}
-			${this.#renderButton()}
-		`;
+		return html`${repeat(
+			this._mappedGroups,
+			(group) => group.key + group.blocks,
```

Comment:
I think + group.blocks would stringify it into "[array Array]". So in this cause its properly better to use the `group.blocks.length`?  but I think it would be better to just base this of the `group.key`. Lets try to make just that work.